Python/gc_free_threading.c

```diff
@@ -1159,7 +1159,8 @@ gc_collect_main(PyThreadState *tstate, int generation, _PyGC_Reason reason)
     if (generation+1 < NUM_GENERATIONS) {
         gcstate->old[generation].count += 1;
     }
-    gcstate->young.count = 0;
+
+    _Py_atomic_store_int(&gcstate->young.count, 0);
```

Comment:
Instead of using atomic operations here, I think we should move the updates to the counters to within the first stop-the-world pause in `gc_collect_internal`.

Reply:
@colesbury thanks - done - that looks cleaner to be honest.